### ISSUE TITLE
CentOS to XCP-ng

### ISSUE THREAD

**Issue by olivierlambert** [xcp-ng/xcp#4]
Using RPMs in XenServer ISO, we must be able to get a clean CentOS 7 to XCP. And to have a working environment (ie: XAPI running, can connect to XO or using xe)

**Comment by Storycrafter** [comment]
- [ ] Determine which Centos 7 as base
- [ ] Determine minimal Centos 7 set of packages and create group for base_xenserver or somesuch
- [ ] Gather extra Xenserver special sauce RPM's from XenServer disk
- [ ] Generate list of locations required to change for custom labeling
- [ ] Locate potential CI/CD type builder to generate iso's

**Comment by olivierlambert** [comment]
* `v6d` is the Citrix licensing daemon. We are modifying XAPI to be able to work without it (right now, XAPI isn't able to work if it's not there).
* `vendor-update-keys-1.3.3-2.noarch.rpm` are keys from various vendors regarding signed XS packs
* others are mainly drivers I suppose?
* `xs-clipboardd-0.0.2-1`: what's that?
* `update-XS73-1.0-1.noarch.rpm` maybe something to update the distro name

**Comment by johnelse** [comment]
pbis-open is this: https://github.com/BeyondTrust/pbis-open

It used to be called Likewise, and is used for Active Directory integration.

**Comment by olivierlambert** [comment]
Okay so let's make 3 categories:

1. mandatory packages (must be inside XCP-ng in order to work)
2. optional packages (like `pbis-open`)
3. packages to avoid (`v6d)`

**Comment by necouchman** [comment]
> xs-clipboardd-0.0.2-1

Not 100% certain, but would venture a guess that this enables copy/paste between a console session and a virtual machine.  We probably want it, if possible and the license allows it.

**Comment by olivierlambert** [comment]
@johnelse does `xs-clipboardd` ring any bell?

**Comment by johnelse** [comment]
> @johnelse does xs-clipboardd ring any bell?

No, but I suspect @necouchman's guess is right. Judging by the version number it's very new :-)

**Comment by koumdros** [comment]
ok, I;m 3 months late and you might have already solved all these , seeing XCP-Ng is out and about but here goes:
elxocmcore
elxocmcorelibs
elxocmlibhbaapi

have to do with emulex HBAs. mostly fibre channel drivers and support files. there should be another utility there, the one used to manipulate their settings. named hbacmd 

I was directed by support personnel to use those to troubleshoot some connections in some of our old servers a couple years back and the name popped .

**Comment by olivierlambert** [comment]
Hi!

XCP-ng is more based on XenServer than directly on CentOS. So it's still discussed on how to get XCP-ng from a bare CentOS. I'm afraid it's not a priority here for us, but any community effort is obviously welcome!

Thanks!

**Comment by koumdros** [comment]
I sort of find your answer lacking consistency :)

ok, it might not be clear I knew that xcp-ng was a XS derivative, I give you that ( I do..)

But, you did open this issue yourself, and you particiapted in the discussion here, so it was something you were interested in, so the last sentence semi-dismissing what i wrote seems..perplexing.

anyway.
I believe this is indeed useful, as it will allow more options beyond upgrading to xcp-ng from centos:

A. keep track of what packages Citrix / Xenserver picks from upstream providers ( mostly drivers and control applications for NIC / HBAs / storage )
B. keep track of related packages / files Citrix might have omitted from said packages ( they sometimes do not include the control utilities for said drivers)
C. keep track what packages Citrix changes from the ones in mainstream Centos 7.X ( RHEL further up), these to my experience are very few.

A.1 would allow to keep a "new-experimental-drivers" secondary yum repo with newer versions of drivers as they are provided by hardware manufacturers, these will probably be included in a "next" version of XS but might solve problems for an XCP-ng user "now" .This has the potential to break things as these wont be as tested but it might be very usefull.

A.2 in the same or a different "extras" repo provide the stripped control utilities. These are 99,9% backward compatible as they almost always have to play with past and present drivers, and are a usefull addition, and sometimes needed to troubleshoot issues.
( note: i have been instructed to install these kind of files from two different HW support teams, and they saved my bacon )

B. this could provide insight on why Citrix did just right that ( incompatibilities? / mere desire for less possible "surface" for problems?)

C. ditto for this, if we find a package is changed and incompatible , it might provide insight and / or options. ( maybe a newer upstream version fixes the problem that pushed Citrix to patch / omitt it in XS?)
enter possible 3 repo "updated-core-packages"

Bonus: All these also lent to the mentality I'm sensing you;re promoting, a more "natural" approach to upgrading and adding extra features to xcp-ng ( plus compatible to your idea for a wrapper around yum commands for XO)

**Comment by koumdros** [comment]
QCS-CLI is the driver for Qlogic nics and HBAs,

QConvergeConsoleCLI-Citrix is a Citrix -specific version of the control utility. there is a newer version floating out there, that as far as i can tell hasn't been "Citrix-ized".

**Comment by stormi** [comment]
Thanks to all who worked on this issue until now. I'm joined the team yesterday and I'm going through the list of RPMs, especially those without source RPMs, and the information provided here is useful :)

**Comment by stormi** [comment]
Hello. I did not comment on this issue since I joined as Release Manager so I suppose I'll do it now. Since this issue was first created, we learned a lot about how XenServer (and so XCP-ng) are structured, what they are made of, and in fact there's no easy way to turn an existing CentOS into an XCP-ng host. And would not bring many benefits, since XCP-ng hosts have to be dedicated to XCP-ng and there's already an installer taking care of all the details and optimizations. XCP-ng is not just a matter of running Xen and VMs. There are many storage, network and management layers that need to work together in a consistent way, and we need full control over what the dom0 contains and how it is configured if we want to guarantee that things work as expected.

However, we did work on making the development and build of the RPMs more open and use an open build system. Take a look at https://github.com/xcp-ng/xcp/wiki/Development-process-tour for details about that.
